src/waveform/renderers/waveformrenderbeat.cpp

```diff
@@ -89,13 +91,25 @@ void WaveformRenderBeat::draw(QPainter* painter, QPaintEvent* /*event*/) {
             m_beats.resize(m_beats.size() * 2);
         }
 
+        auto* waveformBeat = &m_beats[beatCount];
+        waveformBeat->setPosition(xBeatPoint);
+        waveformBeat->setType(beat.getType());
+        waveformBeat->setBeatGridMode(m_waveformRenderer->beatGridMode());
+
         if (orientation == Qt::Horizontal) {
-            m_beats[beatCount++].setLine(xBeatPoint, 0.0f, xBeatPoint, rendererHeight);
+            m_beats[beatCount].setLength(rendererHeight);
         } else {
-            m_beats[beatCount++].setLine(0.0f, xBeatPoint, rendererWidth, xBeatPoint);
+            m_beats[beatCount].setLength(rendererWidth);
+            m_beats[beatCount].setOrientation(Qt::Vertical);
```

Comment:
How about moving this out of the if/else block and always explicitly setting the orientation from the waveform's orientation? That way it does not rely on an implicit default value.